torch/autograd/__init__.py

```diff
@@ -111,19 +112,20 @@ def backward(
 
     tensors = (tensors,) if isinstance(tensors, torch.Tensor) else tuple(tensors)
 
+    grad_tensors_: List[Optional[torch.Tensor]]
     if grad_tensors is None:
-        grad_tensors = [None] * len(tensors)
+        grad_tensors_ = [None] * len(tensors)
     elif isinstance(grad_tensors, torch.Tensor):
-        grad_tensors = [grad_tensors]
+        grad_tensors_ = [grad_tensors]
     else:
-        grad_tensors = list(grad_tensors)
+        grad_tensors_ = list(grad_tensors)
 
-    grad_tensors = _make_grads(tensors, grad_tensors)
+    grad_tensors__ = _make_grads(tensors, grad_tensors_)
```

Comment:
I'm not convinced this helps for code readability...
Maybe just name one `grad_tensors` and the other `grad_tensors_filled`/populated ?

Also the result type of `_make_grads` is the same no? Why can't you just re-use the same variable as before?

Reply:
That's a good suggestion, let me rename just that.
And the types are not exactly the same: 
grad_tensors is constructed as list of nullable tensors, but _make_grads returns a tuple.
But let me change code a bit to actually construct tuple here as well.